File: time-conversion/time-conversion.js
//https://www.hackerrank.com/challenges/three-month-preparation-kit-time-conversion/problem?isFullScreen=true&h_l=interview&playlist_slugs%5B%5D=preparation-kits&playlist_slugs%5B%5D=three-month-preparation-kit&playlist_slugs%5B%5D=three-month-week-one

function timeConversion(s) {
    // Write your code here
    const timeList = s.split(':');
    if (timeList[2][2] === 'A' && timeList[0] === '12') {
        timeList[0] = '00'   
    } else if (timeList[2][2] === 'P' && parseInt(timeList[0]) >= 1 && parseInt(timeList[0]) <= 11) {
        timeList[0] = (parseInt(timeList[0]) + 12).toString()
    };
    return timeList.join(':').substring(0, 8)
}

File: time-conversion/time-conversion.py
def timeConversion(s):
    # Write your code here
    s_arr = s.split(':')
    if s_arr[2][2] == 'A' and s_arr[0] == '12':
        s_arr[0] = '00'
    elif s_arr[2][2] == 'P' and int(s_arr[0]) >= 1 and int(s_arr[0]) <= 11:
        s_arr[0] = str(int(s_arr[0]) + 12)
    return ':'.join(s_arr)[0:8]

#cleaner version

def timeConversion(s):
    # Write your code here
    s_arr = s.split(':')
    hour = s_arr[0]
    minute = s_arr[1]
    second = s_arr[2][0:2]
    am_pm = s_arr[2][2:]
    if am_pm == 'AM' and hour == '12':
        hour = '00'
    elif am_pm == 'PM' and int(hour) >= 1 and int(hour) <= 11:
        hour = str(int(hour) + 12)
    return f'{hour}:{minute}:{second}'